File: etablissementAngular/src/app/model/salle-classe.ts
import { Matiere } from "./matiere";

export class SalleClasse {
    private _id!: number;
    private _nom: string;
    private _matieresExclues: Matiere[] = [];
    private _capacite: number;


	constructor(id?: number, nom: string='',  matieresExclues?: Matiere[] , capacite: number=0) {
		
		this._nom = nom;
		this._capacite = capacite;
	}


    /**
     * Getter id
     * @return {number}
     */
	public get id(): number {
		return this._id;
	}

    /**
     * Getter nom
     * @return {string}
     */
	public get nom(): string {
		return this._nom;
	}

  

    /**
     * Getter matieres
     * @return {Matiere[] }
     */
	public get matieresExclues(): Matiere[]  {
		return this._matieresExclues;
	}

    /**
     * Getter capacite
     * @return {number}
     */
	public get capacite(): number {
		return this._capacite;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public set id(value: number) {
		this._id = value;
	}

    /**
     * Setter nom
     * @param {string} value
     */
	public set nom(value: string) {
		this._nom = value;
	}

 

    /**
     * Setter matieres
     * @param {Matiere[] } value
     */
	public set matieresExclues(value: Matiere[] ) {
		this._matieresExclues = value;
	}

    /**
     * Setter capacite
     * @param {number} value
     */
	public set capacite(value: number) {
		this._capacite = value;
	}


}

File: etablissement/src/main/java/com/projetfinal/etablissement/entity/SalleClasse.java
package com.projetfinal.etablissement.entity;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.OneToOne;
import javax.persistence.Version;

@Entity
public class SalleClasse {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Integer id;
	private String nom;
	@ManyToMany(mappedBy = "salles")
    private List<Matiere> matieresExclues;
	private Integer capacite;
//	@OneToOne
//	private Cours cours;
	@Version
	private int version;
	
	public SalleClasse() {
	}

	public SalleClasse(String nom, Integer capacite) {
		this.nom = nom;
		this.capacite = capacite;
	}

	public Integer getId() {
		return id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public List<Matiere> getMatieresExclues() {
		return matieresExclues;
	}

	public void setMatieresExclues(List<Matiere> matieresExclues) {
		this.matieresExclues = matieresExclues;
	}

	public Integer getCapacite() {
		return capacite;
	}

	public void setCapacite(Integer capacite) {
		this.capacite = capacite;
	}

//	public Cours getCours() {
//		return cours;
//	}
//
//	public void setCours(Cours cours) {
//		this.cours = cours;
//	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		SalleClasse other = (SalleClasse) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}
	
}
